core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -2353,25 +2354,47 @@ public List<String> doGetUserListFromPropertiesWithID(String property, String va
             throw new IllegalArgumentException("Filter value cannot be null");
         }
 
+        boolean isOptimizedSearchEnabled = Boolean.parseBoolean(ServerConfiguration.getInstance()
+                .getFirstProperty(JDBCRealmConstants.PROP_ENABLE_OPTIMIZED_JDBC_SEARCH));
+        boolean useOptimizedProcess = isOptimizedSearchEnabled && UID.equals(property);
+
         String sqlStmt;
         if (value.contains(QUERY_FILTER_STRING_ANY)) {
             // This is to support LDAP like queries. Value having only * is restricted except one *.
             if (!value.matches("(\\*)\\1+")) {
                 // Convert all the * to % except \*.
                 value = value.replaceAll("(?<!\\\\)\\*", SQL_FILTER_STRING_ANY);
             }
-            if (!isCaseSensitiveUsername() && UID.equals(property)) {
-                sqlStmt = realmConfig.getUserStoreProperty(JDBCCaseInsensitiveConstants.
-                        GET_USERS_FOR_PROP_WITH_ID_CASE_INSENSITIVE);
+            if (useOptimizedProcess) {
+                if (!isCaseSensitiveUsername()) {
+                    sqlStmt = realmConfig.getUserStoreProperty(JDBCCaseInsensitiveConstants.
```

Comment:
if we use "useOptimizedProcess" and the property is UID only we should search in UM_USER table right?

Reply:
At line 2359 useOptimizedProcess is initialized using both the config and the property being equal to UID.